packages/traceability/issuer-irec-api/test/certificate.e2e-spec.ts

```diff
@@ -0,0 +1,391 @@
+                isPrivate: true
+            })
+            .expect(HttpStatus.CREATED);
+
+        const {
+            body: { success: transferSuccess }
+        } = await request(app.getHttpServer())
+            .put(`/irec/certificate/${id}/transfer`)
+            .set({ 'test-user': TestUser.OrganizationDeviceManager })
+            .send({
+                to: getUserBlockchainAddress(TestUser.OtherOrganizationDeviceManager),
+                amount: certificateTestData.energy
+            })
+            .expect(HttpStatus.OK);
+
+        expect(transferSuccess).to.be.true;
+
+        const {
+            body: { energy: receiverEnergy }
+        } = await request(app.getHttpServer())
+            .get(`/irec/certificate/${id}`)
+            .set({ 'test-user': TestUser.OtherOrganizationDeviceManager })
+            .expect(HttpStatus.OK);
+
+        expect(receiverEnergy.privateVolume).to.equal(certificateTestData.energy);
+    });
+
+    it('should get all certificate events', async () => {
+        const {
+            body: { id }
+        } = await request(app.getHttpServer())
+            .post('/irec/certificate')
+            .set({ 'test-user': TestUser.Issuer })
+            .send(certificateTestData)
+            .expect(HttpStatus.CREATED);
+
+        const { body: events } = await request(app.getHttpServer())
+            .get(`/irec/certificate/${id}/events`)
+            .set({ 'test-user': TestUser.OrganizationDeviceManager })
+            .expect(HttpStatus.OK);
+
+        expect(events.length).to.be.above(0);
+    });
+
+    it('should get certificates without a blockchain account attached', async () => {
+        const { id } = await createCertificate();
+
+        await request(app.getHttpServer())
+            .get(`/irec/certificate/${id}`)
+            .set({ 'test-user': TestUser.UserWithoutBlockchainAccount })
+            .expect(HttpStatus.OK);
+
+        await request(app.getHttpServer())
+            .get(`/irec/certificate`)
+            .set({ 'test-user': TestUser.UserWithoutBlockchainAccount })
+            .expect(HttpStatus.OK);
+    });
+
+    it('should import certificate', async () => {
+        let certificates = await getCertificates(TestUser.OrganizationDeviceManager);
+        expect(certificates.length).to.equal(0);
+
+        const {
+            body: [certificateToImport]
+        } = await request(app.getHttpServer())
+            .get(`/irec/certificate/certificates-to-import`)
+            .set({ 'test-user': TestUser.OrganizationDeviceManager })
+            .expect(HttpStatus.OK);
+
+        expect(certificateToImport.asset).to.equal('test-asset-id');
+
+        await request(app.getHttpServer())
+            .post(`/irec/certificate/import-certificate`)
+            .send({ assetId: certificateToImport.asset })
+            .set({ 'test-user': TestUser.OrganizationDeviceManager })
+            .expect(HttpStatus.CREATED);
+
+        await sleep(10000);
+    });
```

Comment:
Isn't this use case the only one that's I-REC related? Do we need all the tests above? They seem like duplicates to me to the ones in the issuer-api.

Reply:
I've added them to check if all non irec specific stuff works. And some of them failed at that time. So they are required because it helps me not to break non irec functionality

Reply:
Ok. Just not sure why we need these 2 tests:

1. `should create a private certificate`
2. `should transfer a private certificate`

Are private certificates even supported in I-REC?

Reply:
Yep, there are no private/public certificates. Ok, ill remove tests for this, but what about the controller methods? Do I have to remove them too?

Reply:
No, there is no such functionality. Ill remove those tests